qemu/tests/qemu_guest_agent.py

```diff
@@ -1721,6 +1721,101 @@ def gagent_check_path_fsfreeze_hook(self, test, params, env):
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+
+        def log_check(action):
+            msg = "testing %s:%s" % (user_script_path, action)
+            hook_log = session.cmd_output("cat %s" % log_path)
+            if msg not in hook_log.strip().split('\n')[-2]:
+                test.fail("Fsfreeze hook test failed\nthe fsfreeze"
+                          " hook log is %s." % hook_log)
+
+        session = self._get_session(self.params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Checking fsfreeze hook related scripts.",
+                              logging.info)
+        cmd_get_hook_files = "rpm -ql qemu-guest-agent |grep fsfreeze-hook"
+        hook_files = session.cmd_output(cmd_get_hook_files)
+
+        if len(hook_files.strip().split()) != 4:
+            test.fail("Fsfreeze hook files are missed, the output is"
+                      " %s" % hook_files)
+
+        error_context.context("Checking fsfreeze hook path set in config"
+                              " file.", logging.info)
+        cmd_get_hook_path = "cat /etc/sysconfig/qemu-ga | grep" \
+                            " ^FSFREEZE_HOOK_PATHNAME"
+        o_path = session.cmd_output(cmd_get_hook_path)
+        hook_path = o_path.strip().split("=")[1]
+
+        detail = session.cmd_output("ll %s" % hook_path)
+        if not re.search(r".*x.*x.*x", detail):
+            test.fail("Not all users have executable permission"
+                      " of fsfreeze hook, the detail is %s." % detail)
+
+        error_context.context("Checking if agent service is using the"
+                              " fsfreeze hook.", logging.info)
+        cmd_get_hook = "ps aux |grep /usr/bin/qemu-ga |grep fsfreeze-hook"
+        hook_path_info = session.cmd_output(cmd_get_hook).strip()
+        if params['os_variant'] == 'rhel6':
+            error_context.context("For rhel6 guest,need to enable fsfreeze"
+                                  " hook and restart agent service.",
+                                  logging.info)
+            if session.cmd_output(cmd_get_hook) is None:
+                cmd_enable_hook = "sed -i 's/^FSFREEZE_HOOK_ENABLE=0$/" \
+                                  "^FSFREEZE_HOOK_ENABLE=1$/g' %s" % hook_path
+                session.cmd(cmd_enable_hook)
+                session.cmd(params["gagent_restart_cmd"])
+            hook_path_service = hook_path_info.split("--fsfreeze-hook=")[-1]
+        else:
+            hook_path_service = hook_path_info.split("-F")[-1]
+
+        if hook_path_service != hook_path:
+            test.fail("Fsfreeze hook in qemu-guest-agent service is different"
+                      " from config.\nit's %s from service\n"
+                      "it's %s from config." % (hook_path_service, hook_path))
+
+        error_context.context("Create a simple script to verify fsfreeze"
```

Comment:
@vivianQizhu Yes, the following code is creating a user script(individual application-specific scripts) which will be called by hook-script.
The workflow is:
issue fsfreeze/fsthaw cmd -> call fsfreeze-hook script with fsfreeze/fsthaw as argument -> call application-specific scripts(user script) with  fsfreeze/fsthaw as argument 
The application-specific scripts log will be recorded in log file, which will be used to verify if fsfreeze-hook script call user script correctly.